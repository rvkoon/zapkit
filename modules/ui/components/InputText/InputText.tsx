import cx from "classnames";
import { FieldGroup } from "../FieldGroup";
import { ChangeEvent } from "react";
import _ from "lodash";

export interface IInputTextProps {
  className?: string;
  label?: string;
  placeholder: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  type?: string;
  value: string;
  error?: string;
}

export function InputText({
  className,
  label,
  name,
  placeholder,
  onChange,
  type,
  value,
  error,
}: IInputTextProps): JSX.Element {
  return (
    <FieldGroup htmlFor={name} label={label}>
      <input
        type={type}
        name={name}
        placeholder={placeholder}
        onChange={onChange}
        value={value}
        className={cx(
          "min-h-4 bg-primary-blue-extra-light rounded-base py-1 px-2 focus:outline-primary-blue placeholder:text-[14px]",
          error && "shadow-innerCustom shadow-red-200",
          className
        )}
      />
      {error && (
        <p
          className="pl-1 text-red-500 text-[12px]"
          data-testid={`inputText${_.upperFirst(name)}ErrorMessage`}
        >
          {error}
        </p>
      )}
    </FieldGroup>
  );
}
