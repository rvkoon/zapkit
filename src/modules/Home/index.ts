export * from './components';
export * from './helpers';
export * from './hooks';
export * from './services';
export * from './types';
export * from './constants';