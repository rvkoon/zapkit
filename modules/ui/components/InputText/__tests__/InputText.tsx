import { InputText, IInputTextProps } from "../InputText";
import { RenderResult } from "@testing-library/react";
import { render } from "modules/tests";

describe("InputText component", () => {
  const defaultProps: IInputTextProps = {
    placeholder: "placeholder",
    name: "name",
    value: "value",
    onChange: () => null,
  };

  function renderComponent(props: IInputTextProps): RenderResult {
    return render(<InputText {...defaultProps} {...props} />);
  }

  it("should render correctly", () => {
    const { container } = renderComponent({ ...defaultProps });
    expect(container).toMatchSnapshot();
  });
});
