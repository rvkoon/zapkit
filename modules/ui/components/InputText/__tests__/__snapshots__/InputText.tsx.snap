// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText component should render correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-1"
  >
    <input
      class="min-h-4 bg-primary-blue-extra-light rounded-base py-1 px-2 focus:outline-primary-blue placeholder:text-[14px]"
      name="name"
      placeholder="placeholder"
      value="value"
    />
  </div>
</div>
`;
